src/schema_cache.rs

```diff
@@ -0,0 +1,41 @@
+use std::sync::OnceLock;
```

Comment:
this file is unused lol. introduce a rust-side cache is unnecessary right now since we can reuse the `__pydantic_validator/serializer__` entries on the model classes. I was experimenting with how we might implement something like this since we might wind up wanting it in the future if we expand the reuse to cover everything not just `BaseModel`s

Reply:
Interesting, I definitely think a cache would be helpful, especially once we use configs as their own schemas that wrap other schemas, see https://github.com/pydantic/pydantic-core/pull/1341. I'm keen to try to implement this for v2.10.